x/axelarnet/rate_limit.go

```diff
@@ -62,8 +60,22 @@ func (r RateLimiter) RateLimitPacket(ctx sdk.Context, packet ibcexported.PacketI
 	return nil
 }
 
+type RateLimitedICS4Wrapper struct {
+	channel     porttypes.ICS4Wrapper
+	rateLimiter RateLimiter
+	keeper      *keeper.Keeper
```

Comment:
rate limit already has the keeper, do we need it here?

Reply:
I'd rather not couple internal implementation details. The RateLimiter's only API (as far as other objects are concerned) should be `RateLimitPacket`. That means the setup is a little bit more verbose, but also that we can completely change how the rate limiter works and it won't affect the ics4 wrapper or any other part that uses the rate limiter